src/Microsoft.ML.Dnn/ImageClassificationTransform.cs

```diff
@@ -189,6 +189,27 @@ private void CheckTrainingParameters(ImageClassificationEstimator.Options option
             return (jpegData, resizedImage);
         }
 
+        private static Tensor EncodeByteAsString(VBuffer<byte> buffer)
+        {
+            int length = buffer.Length;
+            var size = c_api.TF_StringEncodedSize((UIntPtr)length);
+            var handle = c_api.TF_AllocateTensor(TF_DataType.TF_STRING, IntPtr.Zero, 0, (UIntPtr)((ulong)size + 8));
+
+            IntPtr tensor = c_api.TF_TensorData(handle);
+            Marshal.WriteInt64(tensor, 0);
+
+            var status = new Status();
+            unsafe
+            {
+                fixed (byte* src = buffer.GetValues())
+                    c_api.TF_StringEncode(src, (UIntPtr)length, (sbyte*)(tensor + sizeof(Int64)), size, status);
```

Comment:
I don't understand why we are saying this is a "string" tensor, when it is the bytes of the image. Why is that a `string`?

Reply:
Tensorflow expects the bytes as a string type.